netplan/cli/commands/ip.py

```diff
@@ -90,21 +90,19 @@ def lease_method_ifindex():
                     logging.debug('Cannot read file %s: %s', ifindex_f, str(e))
                     raise
 
-            def lease_method_nm_connection():  # pragma: nocover (covered in autopkgtest)
+            def lease_method_nm_connection():
                 # FIXME: handle older versions of NM where 'nmcli dev show' doesn't exist
                 try:
-                    nmcli_dev_out = subprocess.Popen(['nmcli', 'dev', 'show', self.interface],
-                                                     env={'LC_ALL': 'C'},
-                                                     stdout=subprocess.PIPE)
-                    for line in nmcli_dev_out.stdout:
-                        line = line.decode('utf-8')
+                    nmcli_dev_out = subprocess.check_output(['nmcli', 'dev', 'show', self.interface],
```

Comment:
Not strictly related to the change from this PR but if you want (and if you're targetting a version of NM that has this option) you can use the following construct to simplify how the output is parsed:

```sh
nmcli --get-values GENERAL.CONNECTION dev show "$interface"
nmcli --get-values connection.uuid show id "$conn_id"
```

This is strictly up to you.

Reply:
This is a good suggestion, but I'll leave it open for a bigger refactoring of `ip.py`. For now, I just want to provide a minimal fix to solve the reported bug (LP#1979674).